src/popup/router/pages/Invite.vue

```diff
@@ -0,0 +1,180 @@
+        <button class="invite-link-copy" v-clipboard:copy="referral.link"><CopyIcon /></button>
+      </div>
+      <AmountSend
+        v-if="referral.topUp"
+        @changeAmount="val => (referral.topUpAmount = val)"
+        :value="referral.topUpAmount"
+      />
+      <template v-if="!referral.topUp">
+        <Button half dark @click="referral.topUp = true">{{ $t('pages.invite.top-up') }}</Button>
+        <Button half @click="claim(referral.link)">{{ $t('pages.invite.claim') }}</Button>
+      </template>
+      <template v-else>
+        <Button half dark @click="referral.topUp = false">{{ $t('pages.invite.close') }}</Button>
+        <Button half @click="topUp(referral)">{{ $t('pages.invite.top-up') }}</Button>
+      </template>
+    </div>
+    <Loader v-if="loading" />
+  </div>
+</template>
+
+<script>
+import { mapState, mapGetters } from 'vuex';
+import { Crypto } from '@aeternity/aepp-sdk/es';
+import Button from '../components/Button';
+import AmountSend from '../components/AmountSend';
+import CopyIcon from '../../../icons/copy.svg?vue-component';
+import { aeToAettos } from '../../utils/helper';
+import { formatDate } from '../../utils';
+
+export default {
+  components: { Button, AmountSend, CopyIcon },
+  data: () => ({ amount: 0, referrals: [], loading: false }),
+  filters: { formatDate },
+  computed: {
+    ...mapState(['sdk', 'current']),
+    ...mapGetters('invites', ['allReferrals']),
+  },
+  created() {
+    this.$watch(
+      'allReferrals',
+      val => {
+        this.referrals = [...val];
+      },
+      { immediate: true },
+    );
+    this.$store.dispatch('invites/getBalances');
+  },
+  methods: {
+    async generate() {
+      this.loading = true;
+      await this.$watchUntilTruly(() => this.sdk);
+      const { publicKey, secretKey } = Crypto.generateKeyPair();
+      if (this.amount) {
+        try {
+          await this.sdk.spend(+aeToAettos(this.amount), publicKey, {
+            payload: 'referral',
+          });
+        } catch (e) {
+          this.$store.dispatch('modals/open', { name: 'default', msg: e.message });
+          this.loading = false;
+          return;
+        }
+      }
+      const link = `https://superhero.com/i/${Crypto.encodeBase58Check(
+        Buffer.from(secretKey, 'hex'),
+      )}`;
+
+      this.$store.commit('invites/add', {
+        link,
+        publicKey,
+        secretKey,
+        balance: 0,
+        date: Date.now(),
+        networkId: this.sdk.getNetworkId(),
+      });
+      this.loading = false;
+      this.$store.dispatch('invites/getBalances');
+    },
+    async claim(referral) {
+      this.loading = true;
+      await this.$store.dispatch('invites/claim', referral);
+      this.loading = false;
+      this.$store.dispatch('invites/getBalances');
+    },
+    async topUp({ link, topUpAmount, publicKey }) {
+      this.loading = true;
+      const address = publicKey || this.$store.getters['invites/keypair'](link);
```

Comment:
`invites/keypair` returns an object with `publicKey`
```suggestion
      const address = publicKey || this.$store.getters['invites/keypair'](link). publicKey;
```